xbmc/filesystem/File.cpp

```diff
@@ -387,6 +397,21 @@ bool CFile::OpenForWrite(const CURL& file, bool bOverWrite)
   return false;
 }
 
+bool CFile::ReOpen(const CURL& file)
+{
+  if (!m_pFile)
```

Comment:
how about?
if (!m_pFile || !m_pFile->SupportsReOpen())

virtual bool IFile::SupportsReOpen() { return false;}

implementation of IFile can override this